openwisp_controller/config/static/config/js/widget.js

```diff
@@ -172,6 +172,12 @@
                 }
             });
         }
+        // show config errors if it is present and hidden
+        var config_field  = $('.field-config');
+        var config_error = config_field.find('.errorlist');
+        if (config_error.length && config_field.css('display') == 'none'){
+            $(config_error).insertAfter('#flat-json-default_values');
+        }
```

Comment:
I don't understand what you're doing here.
This file contains logic that is specific to the JSON schema UI generator widget, it's reused in different parts of OpenWISP (Device Confg, VPN, Template, Credentials), I see you're calling `flat-json-default_values` which is specific to `Template` so this code shouldn't be put here.

I would not touch this part, I don't feel it's necessary, what are you trying to solve? What is the problem, what is the improvement? Can you share a screenshot please?

Reply:
I used `ValidationError({'config': _('This field is required.')})` because i wanted the error to be shown above the config field for easy use. but as shown below, without this `js code`, the error message is not display on UI because the `config-field` has `display set to none`. 
![Screenshot](https://user-images.githubusercontent.com/24882630/94166709-90b25a80-fe83-11ea-82b1-69050052beec.png)

Reply:
Since this file (widget.js) is used by so many other modules which has nothing to do with this solution as you have pointed out. I will use `ValidationError(_('The configuration field is required.'))` which will display the error message at the top. unless you have another suggestion which you will love I implement. :smile: